test/cpp/end2end/grpclb_end2end_test.cc

```diff
@@ -406,17 +424,29 @@ class GrpclbEnd2endTest : public ::testing::Test {
   void ResetStub(int fallback_timeout = 0,
                  const std::string& expected_targets = "",
                  int subchannel_cache_delay_ms = 0) {
+    // Send a separate user agent string for the grpclb load balancer alone.
+    grpc_core::ChannelArgs grpclb_channel_args;
+    // Set a special user agent string for the grpclb load balancer. It
+    // will be verified at the load balancer.
+    grpclb_channel_args = grpclb_channel_args.Set(
+        GRPC_ARG_PRIMARY_USER_AGENT_STRING, kGrpclbSpecificUserAgentString);
     ChannelArguments args;
     if (fallback_timeout > 0) args.SetGrpclbFallbackTimeout(fallback_timeout);
     args.SetPointer(GRPC_ARG_FAKE_RESOLVER_RESPONSE_GENERATOR,
                     response_generator_.get());
     if (!expected_targets.empty()) {
       args.SetString(GRPC_ARG_FAKE_SECURITY_EXPECTED_TARGETS, expected_targets);
+      grpclb_channel_args = grpclb_channel_args.Set(
+          GRPC_ARG_FAKE_SECURITY_EXPECTED_TARGETS, expected_targets);
     }
     if (subchannel_cache_delay_ms > 0) {
       args.SetInt(GRPC_ARG_GRPCLB_SUBCHANNEL_CACHE_INTERVAL_MS,
                   subchannel_cache_delay_ms * grpc_test_slowdown_factor());
     }
+    auto grpclb_c_channel_args = grpclb_channel_args.ToC().release();
```

Comment:
If you don't call `.release()` here, then you don't need to explicitly call `grpc_channel_args_destroy()` on it later.  And you can pass `grpclb_c_channel_args.get()` to `args.SetPointerWithVtable()`.

Reply:
Fixed.